azure-kusto-ingest/azure/kusto/ingest/descriptors.py

```diff
@@ -37,11 +43,31 @@ def __init__(self, path, size=0, source_id=None):
         self.stream_name = os.path.basename(self.path)
 
         if size == 0:
-            raw_size = int(os.path.getsize(self.path))
-            if self.path.endswith(".gz") or self.path.endswith(".zip"):
-                # TODO: this can be improved by reading last 4 bytes
-                raw_size *= 11
-            self.size = raw_size
+            self._detect_size()
```

Comment:
we try to avoid doing heavy lifting in ctors... let move this closer to usage (or hide it under a property function like .size)

Reply:
Moved to property